DM-53143: Output lists of ingested datasets

dhirving opened the pull request:
After ingesting a batch of datasets, the service now writes a file containing the list of dataset IDs that were ingested.  It also writes a Kafka message pointing to the file.  These will be read by the Prompt Publication Service to get the list of datasets that should be unembargoed.

dhirving committed:
Move top-level code into a class

This will make it less awkward to write helper functions that need access to multiple variables.

dhirving committed:
Output lists of ingested datasets

After ingesting a batch of datasets, we now write a file containing the list of dataset IDs that were ingested.  These will be read by the Prompt Publication Service to get the list of datasets that should be unembargoed.

dhirving committed:
Add kafdrop to test setup

Add an instance of kafdrop to the test setup, to allow inspection of Kafka messages.

dhirving committed:
Write kafka message after datasets ingested

Write a message to Kafka for each batch of datasets ingested, pointing to the file containing the list of datasets in the batch.

dhirving committed:
Update base container

dspeck1 reviewed (approved):
LGTM.  Nice use of transactions.  One comment on producer config.

dhirving committed:
Add extra feedback on Kafka status

dhirving committed:
Add missing __init__ that was messing up dev tooling

Add a missing __init__.py that was causing typechecking in unit tests to not work in VSCode

dhirving committed:
Add environment variable for kafka debugging

dhirving committed:
Add sanity check for produce

If auth for the topic is misconfigured, it seems that confluent_kafka will not consistently raise an error when trying to produce to it, and we can thus end up committing consumer offsets without their corresponding outputs.

dhirving commented on python/lsst/queued_butler_writer/kafka.py:
```diff
@@ -22,54 +22,75 @@
+        self._consumer.subscribe([input_topic])
         self._pending_messages: list[Message] = []
 
+        producer_config = common_config | {"transactional.id": "butler-writer-output"}
```
Per the documentation `acks: all` is the default, and it will raise an error if you provide `transactional.id` with anything other than `acks: all`, so I think that's OK.  And the default for `retries` is unlimited, which is OK as well since we can't proceed with anything else until the Kafka message is produced.
